engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -790,6 +791,70 @@ describe('System test: QuerySparql', () => {
     });
   });
 
+  describe('Comunica sparql regression test for #1029', () => {
+    it('Supports EXISTS sparql keyword', async() => {
+      const sparql = new QueryEngine();
+      const store = new n3.Store();
+      const subject = n3.DataFactory.namedNode('http://subject');
+      const predicate = n3.DataFactory.namedNode('http://predicate');
+      const object = n3.DataFactory.namedNode('http://object');
+      const blank = n3.DataFactory.blankNode();
+      const spo = n3.DataFactory.quad(subject, predicate, object);
+      const bpo = n3.DataFactory.quad(blank, predicate, object);
+      store.addQuads([ spo, bpo ]);
+      const quadStream = await sparql.queryQuads(`
+      CONSTRUCT {
+        ?s ?p ?o.
+      }
+      WHERE { 
+        ?s ?p ?o.
+        OPTIONAL {
+          FILTER EXISTS {
+            ?s2 ?p ?o.
+            FILTER (?s != ?s2)
+          }
+        }
+      }
+      `,
+      { sources: [ store ]});
+      const result_quads = await quadStream.toArray();
+      expect(result_quads).toEqual([{
```

Comment:
Could the quads and terms be constructed using the data factory (like in the other tests?)

Reply:
absolutely, next step for me today!

Reply:
close!!

Reply:
should be addressed